tests/Task.test.ts

```diff
@@ -77,6 +77,48 @@ describe('parsing', () => {
         expect(task!.tags[1]).toEqual('#journal/daily');
     });
 
+    it('parses a task from a line and extract tags and not global filter', () => {
+        // Arrange
+        const originalSettings = getSettings();
+        updateSettings({ globalFilter: '#task' });
+        const line =
+            '- [x] #task this is a done task #tagone #journal/daily 🗓 2021-09-12 ✅ 2021-06-20';
+        const path = 'this/is a path/to a/file.md';
+        const sectionStart = 1337;
+        const sectionIndex = 1209;
+        const precedingHeader = 'Eloquent Section';
+
+        // Act
+        const task = Task.fromLine({
+            line,
+            path,
+            sectionStart,
+            sectionIndex,
+            precedingHeader,
+        });
+
+        // Assert
+        expect(task).not.toBeNull();
+        expect(task!.description).toEqual(
+            '#task this is a done task #tagone #journal/daily',
+        );
+        expect(task!.status).toStrictEqual(Status.Done);
+        expect(task!.dueDate).not.toBeNull();
+        expect(
+            task!.dueDate!.isSame(moment('2021-09-12', 'YYYY-MM-DD')),
+        ).toStrictEqual(true);
+        expect(task!.doneDate).not.toBeNull();
+        expect(
+            task!.doneDate!.isSame(moment('2021-06-20', 'YYYY-MM-DD')),
+        ).toStrictEqual(true);
+        expect(task!.tags.length).toEqual(2);
+        expect(task!.tags[0]).toEqual('#tagone');
+        expect(task!.tags[1]).toEqual('#journal/daily');
+
+        // Cleanup
+        updateSettings(originalSettings);
+    });
```

Comment:
In summary, pruning back and simplifying `parses a task from a line and extract tags and not global filter`, it can become something like this:

```typescript
    it('parses a task from a line and extract tags and not global filter', () => {
        // Arrange
        const originalSettings = getSettings();
        updateSettings({ globalFilter: '#task' });

        const line = '- [x] #task this is a done task #tagone #journal/daily';

        // Act
        const task = constructTaskFromLine(line);

        // Assert
        expect(task!.description).toEqual(
            '#task this is a done task #tagone #journal/daily',
        );

        const expectedTags = ['#tagone', '#journal/daily'];
        expect(task!.tags).toStrictEqual(expectedTags);

        // Cleanup
        updateSettings(originalSettings);
    });
```

Reply:
Where `constructTaskFromLine()` is:

```typescript
function constructTaskFromLine(line: string) {
    const task = Task.fromLine({
        line,
        path: 'file.md',
        sectionStart: 0,
        sectionIndex: 0,
        precedingHeader: '',
    });
    return task;
}
```

Reply:
Covered as part of test refactor.